### PR TITLE
Add Multiplay + Matchmaker to example scene

### PR DESCRIPTION
**Pull request by picrossin**
This PR adds an example scene `TestNetcodeExample/Main`, which shows how to connect to Multiplay.

I have also set up our Multiplay and Matchmaker services and the scripts necessary to interact with them, mainly `ServerStartup.cs` and `MatchmakerClient.cs`

After this goes in I will integrate these scripts into our actual game. None of this touches any of the existing work

### CODE REVIEW

File: Eggstravaganza/Assets/TestNetcodeExample/_Scripts/Player/PlayerTransform.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    private void Awake() {
+        _rb = GetComponent<Rigidbody>();
+
+        var permission = _serverAuth ? NetworkVariableWritePermission.Server : NetworkVariableWritePermission.Owner;
+        _playerState = new NetworkVariable<PlayerNetworkState>(writePerm: permission);
+    }
+
+    public override void OnNetworkSpawn() {
+        if (!IsOwner) Destroy(transform.GetComponent<TestPlayerController>());
```
Comment (at669) [inline Eggstravaganza/Assets/TestNetcodeExample/_Scripts/Player/PlayerTransform.cs:26]: just curious, but did you ever run into issues with this line? when i was trying this in a different scene/branch, Destroying gave me non-matching objects across clients (components mismatch etc) so i ended up setting enabled = false, but yours works!

File: Eggstravaganza/Assets/Scripts/MatchmakerClient.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    public void StartClient()
+    {
+        CreateTicket();
+    }
+
+    private async void SignIn()
+    {
+        await ClientSignIn("Gopher");
```
Comment (at669) [inline Eggstravaganza/Assets/Scripts/MatchmakerClient.cs:36]: nice name

File: Eggstravaganza/Assets/Scripts/MatchmakerClient.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            {
+                case MultiplayAssignment.StatusOptions.Found:
+                    gotAssignment = true;
+                    TicketAssigned(multiplayAssignment);
+                    break;
+                case MultiplayAssignment.StatusOptions.InProgress:
+                    break;
+                case MultiplayAssignment.StatusOptions.Failed:
+                    gotAssignment = true;
+                    Debug.LogError($"Failed to get ticket status. Error: {multiplayAssignment.Message}");
```
Comment (at669) [inline Eggstravaganza/Assets/Scripts/MatchmakerClient.cs:114]: is this where a future TODO will do a "retry up to X times" loop?

File: Eggstravaganza/Assets/Scripts/ServerStartup.cs [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    private async void Start()
+    {
+        var server = false;
+        var args = Environment.GetCommandLineArgs();
+        for (var i = 0; i < args.Length; i++)
+        {
+            if (args[i] == "-dedicatedServer")
+            {
+                server = true;
+            }
```
Comment (at669) [inline Eggstravaganza/Assets/Scripts/ServerStartup.cs:45]: just for my understanding, so when you're launching on multiplay, you are literally running a linux build with this flag in some field in the multiplay settings to start a server?

File: Eggstravaganza/Assets/Scripts/ServerStartup.cs [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+        m_Backfilling = true;
+#pragma warning disable 4014
+        BackfillLoop();
+#pragma warning restore 4014
+    }
+
+    private async Task BackfillLoop()
+    {
+        while (m_Backfilling && NeedsPlayers())
```
Comment (at669) [inline Eggstravaganza/Assets/Scripts/ServerStartup.cs:217]: prob for our game we can just do a little `&& !inGame` kind of thing if that might be easier than changing the existing game logic?

**Review by at669 — APPROVED** [review]
skimmed a lot of tarodev's scripts, and mostly reviewed this to better my own understanding. tried it again this morning and it worked for three editor instances for me!

File: Eggstravaganza/Assets/Scripts/ServerStartup.cs (hunk 4 above)
Comment (picrossin) [reply Eggstravaganza/Assets/Scripts/ServerStartup.cs:45]: Yep yep exactly, it's in the build config! I also tell it to use batchmode and no rendering there

File: Eggstravaganza/Assets/Scripts/MatchmakerClient.cs (hunk 3 above)
Comment (picrossin) [reply Eggstravaganza/Assets/Scripts/MatchmakerClient.cs:114]: Yeah that seems wise...

**Pull request merged**